pandas/core/apply.py

```diff
@@ -396,6 +394,16 @@ def agg_list_like(self) -> FrameOrSeriesUnion:
                     "cannot combine transform and aggregation operations"
                 ) from err
             return result
+        else:
+            # Concat uses the first index to determine the final indexing order.
```

Comment:
this is still extra complicated, can we not simply union all the indexes?

Reply:
I think a union is only possible between two indices, not all at once? That would mean creating a loop over all indices, and I dont see how that is less complicated.
 Also, that would not work in the following case:
```python3
 # good index: ["foo", "bar", "lorem", "ipsum"]
>>> id1 = pd.Index(["bar", "lorem", "ipsum", "foo"]) # foo missing
>>> id2 = pd.Index(["foo", "lorem", "ipsum", "foo"]) # bar missing
>>> id3 = pd.Index(["foo", "bar", "lorem", "ispum"]) # correct index
>>> id1.union(id2).union(id3)
Index(['bar', 'foo', 'ipsum', 'ispum', 'lorem'], dtype='object')
>>> id1.union(id2).union(id3)
Index(['bar', 'foo', 'ipsum', 'ispum', 'lorem'], dtype='object')
```

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.union.html#pandas.Index.union

Reply:
cc @rhshadrach

Reply:
+1 on the change here - I don't see a better way. If there was a well-defined way to union the indices, it seems to me this would still be more performant.